src/component/isolation.rs

```diff
@@ -93,46 +97,133 @@ impl ProcessIsolationController for PythonUnsafeController {
     fn boot_process(&self) -> Result<Box<dyn IsolatedProcessHandle>, WorkerError> {
         let pipe = self.pipe_creator.new_pipe()?;
 
-        let c_in = canonicalize(pipe.component_input_file())?.into_os_string();
-        let c_out = canonicalize(pipe.component_output_file())?.into_os_string();
+        let c_in = canonicalize(pipe.component_input_file())?
+            .into_os_string()
+            .into_string()
+            .map_err(WorkerErrorKind::OsStringConversion)?;
+        let c_out = canonicalize(pipe.component_output_file())?
+            .into_os_string()
+            .into_string()
+            .map_err(WorkerErrorKind::OsStringConversion)?;
 
         let subprocess = Popen::create(
-            &[
-                "python3",
-                "-u",
-                &self.executable_file,
-                &c_in.to_string_lossy(),
-                &c_out.to_string_lossy(),
-            ],
+            &["python3", "-u", &self.executable_file, &c_in, &c_out],
             PopenConfig::default(),
         )?;
 
-        Ok(Box::new(PythonUnsafeHandle { subprocess, pipe }))
+        Ok(Box::new(PipedProcessHandle { subprocess, pipe }))
     }
 }
 
 #[derive(Debug)]
-pub struct PythonUnsafeHandle {
+struct DockerArchiveController {
+    pipe_creator: NamedPipeCreator,
+    docker_image_tag: String,
+}
+
+impl DockerArchiveController {
+    pub fn new(docker_tar_file_path: &str) -> Result<Self, WorkerError> {
+        if cfg!(target_os = "macos") {
+            warn!("using docker for isolation on macOS likely will not work!!!");
+        }
```

Comment:
This may or may not be terrible. I'm not sure what the right thing to do is. @pcodes ?